students/kate/session05/mailroom3.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+"""
+Mailroom Exercise, session 5
+exception handling, comprehensions
+"""
+import math
+
+# define global donor dict and $
+donors_dict = {
+    'Hermione Granger': [43.01, 13.17],
+    'Molly Weasley':[87.03],
+    'Luna Lovegood':[61.03, 44.87, 27.77],
+    'Sybill Trelawney':[77.56, 33.45,  756.32]
+    }
+
+# create a function for the main menu
+def main_menu():
+    action = input('''
+      Thank you donating to Hogwarts School. Choose an action:
+
+      1. - Send a Thank You Owl
+      2. - Create a Report
+      3. - Quit
+
+      > ''')
+    return action.strip()
+
+def write_thankyou(donor_input, amount):
+    return('''
+          Dear {}
+
+          Thank you for your very donation of ${:.2f}.
+          You have enabled excellemt magical instruction!
+
+                         Sincerely,
+                            -Hogwarts School of Witchcraft and Wizardy
+          '''.format(donor_input, amount))
+
+def send_thankyou():
+
+
+    while True:
+        donor_input = input('''
+        Enter a donor's name.
+
+        *or*
+        Select 1. - View donor list
+        Select 2. - Return to main menu
+
+        >
+        ''').strip()
+
+        if donor_input == '1':
+                print_donors_list()
+        elif donor_input == '2':
+            return
+        else:
+            break
+
+    while True:
+        amount = input("Enter a donation amount > ").strip()
+        if amount == "quit":
+            return
+        try:
+            amount = float(amount)
+        except ValueError:
+            print("Invalid entry! Please enter a number!")
+        else:
+            break
+
+    donors_dict[donor_input] = amount
```

Comment:
Here's you issue ( I think :-) ):

amount is a float -- perfect type for, well, an amount of something :-)
but donors_dict is a dict with the names as the keys, and a LIST of amounts as the value.

With this line, you are replacing a list of values with a single value -- which is going to cause problems later when you need to work with that list (plus you will have thrown away any previous donations)

You want: 

`donors_dict[donor_input].append(amount)`

if this is a new donor, you want to have put an empty list in there before this line.